src/main/kotlin/io/snyk/plugin/ui/jcef/OpenFileLoadHandlerGenerator.kt

```diff
@@ -3,29 +3,33 @@ package io.snyk.plugin.ui.jcef
 import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.editor.colors.ColorKey
 import com.intellij.openapi.editor.colors.EditorColorsManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFileManager
 import com.intellij.ui.JBColor
 import com.intellij.ui.jcef.JBCefBrowserBase
 import com.intellij.ui.jcef.JBCefJSQuery
 import com.intellij.util.ui.UIUtil
 import io.snyk.plugin.getDocument
 import io.snyk.plugin.navigateToSource
-import io.snyk.plugin.SnykFile
 import org.cef.browser.CefBrowser
 import org.cef.browser.CefFrame
 import org.cef.handler.CefLoadHandlerAdapter
 import java.awt.Color
+import java.nio.file.Paths
 
-class OpenFileLoadHandlerGenerator(snykFile: SnykFile) {
-    private val project = snykFile.project
-    private val virtualFile = snykFile.virtualFile
-
+class OpenFileLoadHandlerGenerator(private val project: Project) {
     fun openFile(value: String): JBCefJSQuery.Response {
         val values = value.replace("\n", "").split(":")
+        val filePath = values[0]
         val startLine = values[1].toInt()
         val endLine = values[2].toInt()
         val startCharacter = values[3].toInt()
         val endCharacter = values[4].toInt()
 
+        val virtualFile =
+            VirtualFileManager.getInstance().findFileByNioPath(Paths.get(filePath))
+                ?: return JBCefJSQuery.Response("error")
+
```

Comment:
Is there a chance to pre-calculate the VirtualFile? IntelliJ doesn't like to do this kind of thing in the UI thread and will log a warning/error if it takes too long.

Reply:
Good suggestion, I think I did that in https://github.com/snyk/snyk-intellij-plugin/pull/550/commits/cbe985923cc1ac75f2ab25b0f2673bf275428436. Basically just pre-computing the virtual files before building the query used by the UI